### PR TITLE
When an invocation is not send, this will check if someone else dereg…

### PR DESCRIPTION
**Pull request by sancar**
…istred the invocation so that there will be no spurious retry.

### CODE REVIEW

File: hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java [hunk 1]
```diff
@@ -96,10 +96,17 @@ protected void send(ClientInvocation invocation, ClientConnection connection) th
         ClientMessage clientMessage = invocation.getClientMessage();
         if (!isAllowedToSendRequest(connection, invocation) || !writeToConnection(connection, clientMessage)) {
             final int callId = clientMessage.getCorrelationId();
```
Comment (pveentjer) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java]: are call id's not a long?

Comment (sancar) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java]: No, they are not. I just checked operation call id it seems they are long, but as far as I remember client call ids were always int.

**Comment by mdogan** [comment]
looks good... I wish you were able to write a test.. :+1:

**Comment by sancar** [comment]
verify

**Comment by bilalyasar** [comment]
run-parallel

**Comment by bilalyasar** [comment]
run-parallel

File: hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java [hunk 2]
```diff
@@ -96,10 +96,17 @@ protected void send(ClientInvocation invocation, ClientConnection connection) th
...
-            deRegisterCallId(callId);
+            ClientInvocation clientInvocation = deRegisterCallId(callId);
             deRegisterEventHandler(callId);
-            throw new IOException("Packet not send to " + connection.getRemoteEndpoint());
+            if (clientInvocation != null) {
+                throw new IOException("Packet not send to " + connection.getRemoteEndpoint());
+            } else {
+                if (logger.isFinestEnabled()) {
```
Comment (ihsandemir) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java]: Do we understand why such a case occurs? Who deregisters the call?

**Comment by ihsandemir** [comment]
OK :+1: Let's change the correlationId to "long" in another PR (new protocol defines correlationId to be uint32).

**Pull request merged**